source/persistency/PetaloPersistencyManager.cc

```diff
@@ -285,114 +284,49 @@ void PetaloPersistencyManager::StoreIonizationHits(G4VHitsCollection* hc)
 
 void PetaloPersistencyManager::StoreSensorHits(G4VHitsCollection* hc)
 {
-  std::map<G4int, SensorHit*> mapOfHits;
-
-  SensorHitsCollection* hits = dynamic_cast<SensorHitsCollection*>(hc);
+  PetSensorHitsCollection* hits = dynamic_cast<PetSensorHitsCollection*>(hc);
   if (!hits) return;
 
-  std::vector<G4int > sensor_ids;
   for (size_t i=0; i<hits->entries(); i++) {
 
-    SensorHit* hit = dynamic_cast<SensorHit*>(hits->GetHit(i));
+    PetSensorHit* hit = dynamic_cast<PetSensorHit*>(hits->GetHit(i));
     if (!hit) continue;
 
-    int s_id  = hit->GetPmtID();
-    mapOfHits[s_id] = hit;
-
-    const std::map<G4double, G4int>& wvfm = hit->GetHistogram();
-    std::map<G4double, G4int>::const_iterator it;
+    G4int s_id   = hit->GetSnsID();
+    G4int charge = hit->GetDetPhotons();
 
-    G4double amplitude = 0.;
-    for (it = wvfm.begin(); it != wvfm.end(); ++it) {
-      amplitude = amplitude + (*it).second;
-    }
-    if (hit->GetPmtID() >= 0) {
-      G4int sens_id;
-      sens_id = hit->GetPmtID();
-
-      if (amplitude > thr_charge_){
-        sensor_ids.push_back(sens_id);
-      }
-    } else if (hit->GetPmtID()<0) {
-      tof_bin_size_ = hit->GetBinSize();
-    }
-  }
-
-  for (G4int s_id: sensor_ids){
-    std::string sdname = hits->GetSDname();
-
-    SensorHit* hit = mapOfHits[s_id];
-    G4ThreeVector xyz = hit->GetPosition();
-
-    const std::map<G4double, G4int>& wvfm = hit->GetHistogram();
-    std::map<G4double, G4int>::const_iterator it;
-
-    if (save_tot_charge_ == true) {
-      G4double charge = 0.;
-      for (it = wvfm.begin(); it != wvfm.end(); ++it) {
-        charge = charge + (*it).second;
+    if (charge > thr_charge_){
+      std::string sdname = hits->GetSDname();
+      G4ThreeVector xyz = hit->GetPosition();
+      if (save_tot_charge_ == true) {
```

Comment:
Now that the `sns_response` table contains the total charge detected by the SiPM, is it still useful the `save_tot_charge` parameter?

Reply:
I think that this parameter has nothing to do with saving the total charge or the waveform (time bin and charge). We introduced it to be able to avoid writing the table at all (whatever its content) and save space in disk if we are not interested in the sensor response, but maybe I'm wrong.